### PR TITLE
JIT: make global morph its own phase

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Contributes to #93246.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
PTAL @dotnet/jit-contrib 

Once RPO is in place, the block traversal order will change, so split out the per-block action as its own method.

No diffs. Has bugged me forever that there was no default IR dump after morph. Now there is.

**Comment by AndyAyersMS** [comment]
No [diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=455655&view=ms.vss-build-web.run-extensions-tab), errors known.

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -13862,53 +13890,16 @@ void Compiler::fgMorphBlocks()
...
-
-    BasicBlock* block = fgFirstBB;
-    noway_assert(block);
-
-    do
+    // Morph all blocks.
+    //
+    // Note morph can add blocks downstream from the current block,
+    // and alter (but not null out) the current block's bbNext;
+    // this iterator ensures they all get visited.
```
Comment (BruceForstall) [inline src/coreclr/jit/morph.cpp:13897]: Doesn't the iterator load the bbNext before the iteration of the loop? If the callee can adjust the blocks, maybe we shouldn't use the for-based iterator here?

**Review by BruceForstall — APPROVED** [review]

File: src/coreclr/jit/morph.cpp (hunk 1 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/morph.cpp:13897]: It doesn't seem to cache anything...? Also that comment is a bit stale; with other changes I've made recently morph won't add blocks or (I suspect) alter `bbNext` anymore.

Comment (BruceForstall) [reply src/coreclr/jit/morph.cpp:13897]: Maybe it only caches the end-point of the iteration. But still, any change to the block list seems risky to depend on what the C++ compiler is going to do.

Comment (AndyAyersMS) [reply src/coreclr/jit/morph.cpp:13897]: For the `BasicBlockRangeList` form sure... but this is going via `BasicBlockSimpleList` which does't cache an endpoint.

Comment (AndyAyersMS) [reply src/coreclr/jit/morph.cpp:13897]: @BruceForstall still ok with merging? Or would you rather I change the iteration back?

Comment (BruceForstall) [reply src/coreclr/jit/morph.cpp:13897]: Yes, it's fine.

**Pull request merged**